test/IISExpress.FunctionalTests/InProcess/TestServerTest.cs

```diff
@@ -36,5 +37,85 @@ public TestServerTest(ITestOutputHelper output = null) : base(output)
                 Assert.Equal(expectedPath, path);
             }
         }
+
+        [ConditionalFact]
+        [OSSkipCondition(OperatingSystems.Windows, WindowsVersions.Win7, "https://github.com/aspnet/IISIntegration/issues/866")]
+        public async Task WritesSucceedAfterClientDisconnect()
+        {
+            TaskCompletionSource<bool> requestStartedCompletionSource = new TaskCompletionSource<bool>(TaskCreationOptions.RunContinuationsAsynchronously);
+            TaskCompletionSource<bool> requestCompletedCompletionSource = new TaskCompletionSource<bool>(TaskCreationOptions.RunContinuationsAsynchronously);
+
+            var data = new byte[1024];
+            using (var testServer = await TestServer.Create(async ctx => {
+                requestStartedCompletionSource.SetResult(true);
+
+                for (int i = 0; i < 1000; i++)
+                {
+                    await ctx.Response.Body.WriteAsync(data);
+                }
+
+                requestCompletedCompletionSource.SetResult(true);
+            }, LoggerFactory))
+            {
+                using (var connection = testServer.CreateConnection())
+                {
+                    await connection.Send(
+                        "POST / HTTP/1.1",
+                        $"Content-Length: {100000}",
+                        "Host: localhost",
+                        "Connection: close",
+                        "",
+                        "");
+
+                    await requestStartedCompletionSource.Task.TimeoutAfterDefault();
+                }
+
+                await requestCompletedCompletionSource.Task.TimeoutAfterDefault();
+            }
+        }
+
+        [ConditionalFact]
+        [OSSkipCondition(OperatingSystems.Windows, WindowsVersions.Win7, "https://github.com/aspnet/IISIntegration/issues/866")]
+        public async Task ReadThrowsAfterClientDisconnect()
+        {
+            TaskCompletionSource<bool> requestStartedCompletionSource = new TaskCompletionSource<bool>(TaskCreationOptions.RunContinuationsAsynchronously);
+            TaskCompletionSource<bool> requestCompletedCompletionSource = new TaskCompletionSource<bool>(TaskCreationOptions.RunContinuationsAsynchronously);
+
+            Exception exception = null;
+
+            var data = new byte[1024];
+            using (var testServer = await TestServer.Create(async ctx => {
+                requestStartedCompletionSource.SetResult(true);
+                try
+                {
+                    await ctx.Request.Body.ReadAsync(data);
```

Comment:
For my understanding, we expect read to throw on failure and write to silently succeed, correct?

Reply:
Yes